gpii/node_modules/solutionsRegistry/test/SolutionsRegistryFileTests.js

```diff
@@ -65,18 +66,77 @@ gpii.tests.universal.solutionsRegistry.srFiles.checkSingleSolution = function (g
+            if (termDef === undefined) {
+                missingTermsCount++;
+                fluid.log("The capability '" + capabilityKey + "' specified in solution '" + solutionKey + "' does not exist.");
+            }
         });
+
+        jqUnit.assertEquals("There should be no missing capabilities in solution '" + solutionKey + "'.", 0, missingTermsCount);
     }
     else {
         jqUnit.assert("Entry '" + solutionKey + "' has no capabilities, which is fine.");
     }
 };
 
+// Ensure that all transforms involving common terms refer to common terms that actually exist.
+gpii.tests.universal.solutionsRegistry.srFiles.checkTransforms = function (filename, solutionKey, solutionDef, genericTerms) {
+    fluid.each(solutionDef.settingsHandlers, function (settingsHandlerDef, settingsHandlerKey) {
+        var allTransformTerms = [];
+
+        if (settingsHandlerDef.capabilitiesTransformations) {
+            var forwardTransformTerms = gpii.tests.universal.solutionsRegistry.srFiles.extractCommonTerms(settingsHandlerDef.capabilitiesTransformations);
+            if (forwardTransformTerms.length) {
+                allTransformTerms = allTransformTerms.concat(forwardTransformTerms);
+            }
+        }
+
+        if (settingsHandlerDef.inverseCapabilitiesTransformations) {
+            var inverseTransformTerms = gpii.tests.universal.solutionsRegistry.srFiles.extractCommonTerms(settingsHandlerDef.inverseCapabilitiesTransformations);
+            if (inverseTransformTerms.length) {
+                allTransformTerms = allTransformTerms.concat(inverseTransformTerms);
+            }
+        }
+        var missingTermsCount = 0;
+
+        fluid.each(allTransformTerms, function (transformTerm) {
+            var unescapedTermKey = transformTerm.replace(/\\\./g, ".");
+            var termExists = (fluid.get(genericTerms, [unescapedTermKey]) !== undefined);
+
+            if (!termExists) {
+                missingTermsCount++;
+                fluid.log(filename + " -> " + solutionKey + " -> " + settingsHandlerKey + " uses the common term '" + unescapedTermKey + "', which does not exist.");
+            }
+        });
+
+        jqUnit.assertEquals("There should be no transforms that use missing common terms.", 0, missingTermsCount);
+    });
+};
+
+gpii.tests.universal.solutionsRegistry.srFiles.extractCommonTerms = function (toInspect) {
+    var foundTerms = [];
+
+    if (typeof toInspect === "object") {
+        fluid.each(toInspect, function (deepValue) {
+            var subSegments = gpii.tests.universal.solutionsRegistry.srFiles.extractCommonTerms(deepValue);
+            if (subSegments.length) {
+                foundTerms = foundTerms.concat(subSegments);
+            }
+        });
+    }
+    else if (typeof toInspect === "string" && toInspect.match(/http:\/\/registry\\?\.gpii\\?\.net\/common\/.+/)) {
```

Comment:
toInspect.startsWith("http://registry.gpii.net/common/") is a touch less unsightly

Reply:
If you've spent too much time with Perl, every problem looks like an excuse to use regular expressions.  Fixed.